habitat-lab/habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -842,14 +850,19 @@ def step(self, action: Union[str, int]) -> Observations:
             )
             self.add_perf_timing("get_sensor_observations", t_start)
 
-        if self._enable_gfx_replay_save:
+        # TODO: Support recording while batch rendering
+        if self._enable_gfx_replay_save and not self._batch_render:
             self.gfx_replay_manager.save_keyframe()
 
         if self._needs_markers:
             self._update_markers()
 
         # TODO: Make debug cameras more flexible
-        if "third_rgb" in obs and self._debug_render:
+        if (
+            "third_rgb" in obs
+            and self._debug_render
```

Comment:
Can you change this to just set `self._debug_render` to False when `self._batch_render` is enabled? This could catch more debug render situations, rather than just checking this `if` statement.

Reply:
I instead added an assert to `EnvBatchRenderer` to explicit about this.